src/rules/jsxCurlyBracePresenceRule.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Lint from "tslint";
+import { isJsxAttribute, isJsxExpression, isStringLiteral, isTextualLiteral } from "tsutils";
+import * as ts from "typescript";
+
+const OPTION_ALWAYS = "always";
+const OPTION_NEVER = "never";
+const CURLY_PRESENCE_VALUES = [OPTION_ALWAYS, OPTION_NEVER];
+const CURLY_PRESENCE_OBJECT = {
+    enum: CURLY_PRESENCE_VALUES,
+    type: "string",
+};
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "jsx-curly-brace-presence",
+        description: "Enforce curly braces or disallow unnecessary curly braces in JSX props",
+        optionsDescription: Lint.Utils.dedent`
+One of the following two options must be provided:
+
+* \`"${OPTION_ALWAYS}"\` requires JSX attributes to have curly braces around string literal values
+* \`"${OPTION_NEVER}"\` requires JSX attributes to NOT have curly braces around string literal values`,
+        options: {
+            type: "array",
```

Comment:
please use the object-style options syntax ([example](https://github.com/palantir/tslint/blob/e080fc4ceb58e907d48bf5419ceb96e6899dd5d5/src/rules/memberOrderingRule.ts#L182)), so that this gets configured as

```
"jsx-curly-brace-presence": {
  "options": {
    "props": "always" | "never",
  }
}
```

Reply:
Updated to use object instead of array type for the options. The `props` parameter seems to be considered optional by default, then?

Reply:
I am not sure I entirely understand how this options setup works. Do I now need to access the flag as `options.props`? The tests (which I did not change yet) still seem to pass.

Reply:
okay, I think I figured it out. I had expected malformed options break the tests.